File: sCTF-15-q2/fibonacci-not/fib-test.py
glist = [0,1]
wlist = [0,1]

def g(x):
    if (x < len(glist)):
        return glist[x]
    n = (g(x-1) + g(x-2))**2
    glist.append(n)
    return n

def w(x):
    if (x < len(wlist)):
        return wlist[x]
    n = w(x-1)**2 + w(x-2)**2
    wlist.append(n)
    return n

for i in range(3,31):
    print "------g(" + str(i) + ")------\n"
    g(i)
    print glist[i]
    print "------w(" + str(i) + ")------\n"
    w(i)
    print wlist[i]

a = glist[30]
b = wlist[30]
c = a-b
print "g(20):" + str(a)
print "w(20):" + str(b)
print "f(20):" + str(c)

s = 0
tmp = c
while (tmp > 0):
    digit = tmp%10
    s = s + digit
    tmp = tmp / 10

print "sum:" + str(s)

File: sCTF-15-q2/fibonacci-not/fib-test.cpp
#include <iostream>
#include <cmath>
#include <string>
#include <vector>
using namespace std;

long double g(int x, vector<long double>& list) {
    if (x < list.size()) {
	return list[x];
    }
    long double n = pow((g(x-1,list) + g(x-2,list)),2);
    list.push_back(n);
    return n;
}

long double w(int x, vector<long double>& list) {
    if (x < list.size()) {
	return list[x];
    }
    double n = pow(w(x-1,list),2) + pow(w(x-2,list),2);
    list.push_back(n);
    return n;
}

int main() {
    vector<long double> glist = {0,1};
    vector<long double> wlist = {0,1};
    long double a = g(15,glist);
    long double b = w(15,wlist);
    long double c = a-b;
    cout << endl;
    cout << "g(10):" << a << endl;
    cout << "w(10):" << b << endl;
    cout << "f(10):" << c << endl;
    
    long double sum = 0;
    long double tmp = c;
    int digit;
    while (tmp > 0) {
	digit = fmod(tmp,10);
	sum += digit;
	tmp /= 10;
    }
    cout << endl<< "sum: " << sum << endl;
}
